tests/test_api_requests.py

```diff
@@ -322,6 +233,1467 @@ async def parse(self, response):
+        # httpResponseBody and browserHtml are set to False implicitly or
+        # explicitly, httpResponseHeaders is not defined, and a warning is
+        # logged.
+        (
+            {"httpResponseBody": False, "httpResponseHeaders": False},
+            {},
+            ["do not need to set httpResponseHeaders to False"],
+        ),
+        (
+            {"screenshot": True, "httpResponseHeaders": False},
+            {"screenshot": True},
+            ["do not need to set httpResponseHeaders to False"],
+        ),
+        (
+            {
+                "unknownMainOutput": True,
+                "httpResponseBody": False,
+                "httpResponseHeaders": False,
+            },
+            {"unknownMainOutput": True},
+            ["do not need to set httpResponseHeaders to False"],
+        ),
+    ],
+)
+def test_automap_header_output(meta, expected, warnings, caplog):
+    _test_automap({}, {}, meta, expected, warnings, caplog)
+
+
+@pytest.mark.parametrize(
+    "method,meta,expected,warnings",
+    [
+        # The GET HTTP method is not mapped, since it is the default method.
+        (
+            "GET",
+            {},
+            {
+                "httpResponseBody": True,
+                "httpResponseHeaders": True,
+            },
+            [],
+        ),
+        # Other HTTP methods, regardless of whether they are supported,
+        # unsupported, or unknown, are mapped as httpRequestMethod, letting
+        # Zyte API decide whether or not they are allowed.
+        *(
+            (
+                method,
+                {},
+                {
+                    "httpResponseBody": True,
+                    "httpResponseHeaders": True,
+                    "httpRequestMethod": method,
+                },
+                [],
+            )
+            for method in (
+                "POST",
+                "PUT",
+                "DELETE",
+                "OPTIONS",
+                "TRACE",
+                "PATCH",
+                "HEAD",
+                "CONNECT",
+                "FOO",
+            )
+        ),
+        # If httpRequestMethod is also specified in meta with the same value
+        # as Request.method, a warning is logged asking to use only
+        # Request.meta.
+        *(
+            (
+                request_method,
+                {"httpRequestMethod": meta_method},
+                {
+                    "httpResponseBody": True,
+                    "httpResponseHeaders": True,
+                    "httpRequestMethod": meta_method,
+                },
+                ["Use Request.method"],
+            )
+            for request_method, meta_method in (
+                ("GET", "GET"),
+                ("POST", "POST"),
+            )
+        ),
+        # If httpRequestMethod is also specified in meta with a different value
+        # from Request.method, a warning is logged asking to use Request.meta,
```

Comment:
Request.method, not Request.meta?

Reply:
I think the wording is correct, however confusing. It is saying “if Request.meta["httpRequestMethod"] != Request.method”.

Reply:
Hm, but the warning asks to use Request.method, not Request.meta.